lasagne/layers/shape.py

```diff
@@ -21,17 +21,33 @@
 
 class FlattenLayer(Layer):
     """
-    Flatten all but the first dimension.
+    Flatten all dimensions trailing ``dim``
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape.
+    dim : int
+        Dimensions trailing ``dim`` will be flattened
 
     See Also
     --------
     flatten  : Shortcut
     """
+    def __init__(self, incoming, dim=1, **kwargs):
+        super(FlattenLayer, self).__init__(incoming, **kwargs)
+        self.dim = dim
+
+        if dim < 0:
+            raise ValueError('Dim must be >=0, was %s', str(dim))
```

Comment:
Shouldn't this be:
```
raise ValueError('Dim must be >=0, was %d', dim)
```

Or does the `str` serve a purpose? I thought maybe it was to make it raise when someone passes something that isn't a number, but then `dim < 0` would also fail anyway.